lib/resty/core/regex.lua

```diff
@@ -478,6 +479,87 @@ function ngx.re.find(subj, regex, opts, ctx, nth)
     return re_match_helper(subj, regex, opts, ctx, false, nil, nth)
 end
 
+local function destroy_re_gmatch_iterator(iterator)
+    if not iterator._compile_once then
+        destroy_compiled_regex(iterator._compiled)
+    end
+    iterator._compiled = nil
+    iterator._pos = nil
+    iterator._subj = nil
+end
+
+local function iterate_re_gmatch(self)
+    local compiled = self._compiled
+    local subj = self._subj
+    local subj_len = self._subj_len
+    local flags = self._flags
+    local pos = self._pos
+
+    if not pos then
+        -- The iterator is exhausted.
+        return nil
+    end
+
+    local rc = C.ngx_http_lua_ffi_exec_regex(compiled, flags, subj,
+                                             subj_len, pos)
+
+    if rc == PCRE_ERROR_NOMATCH then
+        destroy_re_gmatch_iterator(self)
+        return nil
+    end
+
+    if rc < 0 then
+        destroy_re_gmatch_iterator(self)
+        return nil, "pcre_exec() failed: " .. rc
+    end
+
+    if rc == 0 then
+        if band(flags, FLAG_DFA) == 0 then
+            destroy_re_gmatch_iterator(self)
+            return nil, "capture size too small"
+        end
+
+        rc = 1
+    end
+
+    local cp_pos = tonumber(compiled.captures[1])
```

Comment:
`tonumber` is a global that hasn't been cached locally at the top of this module. Shall we add it?

Reply:
Yeah, good catch.